routes/app/signin.js

```diff
@@ -1,34 +1,50 @@
+const User  = require('../../models/user');
 const { Router }    = require('express');
-
+const passport     = require('passport');
+const bcrypt = require('bcrypt');
 
 const router = new Router();
 module.exports = router;
 
+router.post('/signin', async function(req, res) {
+  const { email, password } = req.body;
 
-router.get('/signin', async function(req, res) {
+  const result = await User.findPass({email, password});
```

Comment:
for this section, try not to nest try/catch statements because it can get confusing

The outermost try/catch should be enough to get all errors. Try this:
```
try {
  // This grabs the first result of the array
  const [ result ] = await User.findPass({ email, password });
  const hash = result.password;
  const compareResponse = await bcrypt.compare(password, hash);

  if(compareResponse){
    await req.login(result.id);
    res.json({
      message: 'success'
    });
  } else {
    res.status(401); // send unauthorized
  }
} catch (error) {
  console.error(error);
}
```

Reply:
The only issue here is that. If the user doesn't exist, result is an empty array, result.password is undefined, and bcrypt hangs when passed an  empty array

Reply:
Oh okay then just have it be:
```
const result = ...
if(result.length === 0) throw new Error('User does not exist');
const user = result[0];
const hash = result[0].password;
```

Reply:
Perfect